### ISSUE TITLE
Social card cyrillic encoding bug

### ISSUE THREAD

**Issue by Zilborg** [squidfunk/mkdocs-material#3099]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

`utf-8` encoding is missing missing (imho).
I think the trouble is missing encoding  in `ImageFont.truetype(self.font.get(700), 36, encoding="utf-8")`, however, I couldn't to fix this. 

Your source [link](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/src/plugins/social/plugin.py#L141-L155).

### Expected behaviour

Correct social card =)

### Actual behaviour

![Screenshot 2021-10-11 at 10 37 09](https://user-images.githubusercontent.com/33390074/136760438-f44a774e-a444-4160-be26-c136615a5ea6.png)

Russian title and description looks like squares.

### Steps to reproduce

1. Write `title = 'Привет мир!'` (Hello world in translation) and/or `description =  'Привет мир!`
2. build 
3. https://cards-dev.twitter.com/validator

### Package versions

Build with Github action.
I have the fork of insiders. There was last build with your hash commit (`96e9a28f989abc498bc6f5ced4d0d061908c95ff`  [commit link](https://github.com/squidfunk/mkdocs-material-insiders/commit/96e9a28f989abc498bc6f5ced4d0d061908c95ff))

### Configuration

```yaml
# Configuration
theme:
  # Default values, taken from mkdocs_theme.yml
  language: en

# Plugins
plugins:
  - social
```

### System information

I think, doesn't matter.

**Comment by squidfunk** [comment]
Thanks for reporting. I think the problem lies in the font that's being downloaded. I'll investigate asap.

**Comment by Zilborg** [comment]
Now it's worked.
![image](https://user-images.githubusercontent.com/33390074/136817469-06163390-e422-426a-af86-61442f1b47e4.png)

Added script to github action. But it's kludge)
``` bash
     ....
      - name: Install Python dependencies
        run: |
          pip install git+https://${GH_TOKEN}@github.com/Zilborg/mkdocs-material-insiders.git
      - name: Download & unzip fonts
        run: |
          curl -o fonts.zip -L https://fonts.google.com/download?family=Roboto
          mkdir .cache
          unzip -p fonts.zip Roboto-Regular.ttf > .cache/Roboto.400.ttf
          unzip -p fonts.zip Roboto-Bold.ttf > .cache/Roboto.700.ttf 
      - name: Deploy documentation
      ....
```

**Comment by squidfunk** [comment]
Fixed in [`351dd162c`](https://github.com/squidfunk/mkdocs-material-insiders/commit/351dd162c3fc3a42fb160e2c008274eeae196182). Thanks again, your efforts were a huge help!

The social plugin will now download the fonts from `https://fonts.google.com/download` and extract them to the `.cache` folder, which ensures that the entire supported character set is available. All fonts are kept, not only the regular and bold weights. This will later allow specifying alternative font weights for different parts of the social card if desired. In the future, the social card will have more configuration options, and this is an important step in that direction 😊

Furthermore, if a font doesn't include a bold variant, the social plugin will not fall back to the regular weight.
